### PR TITLE
[MKL-DNN] Fix to crash of Transformer when mkldnn is to be used

### PR DESCRIPTION
**Pull request by jczaja**
This PR is fixing transformer analyzer test when MKL-DNN is to be used. Root cause of crash was that  TensorCopy was not setting MKLDNN primitive descriptor when layout was to be kMKLDNN

### CODE REVIEW

**Comment by luotao1** [comment]
Please add `TEST(Analyzer_transformer, compare_mkldnn)` and `TEST(Analyzer_transformer, profile_mkldnn)` like `analyer_resnet50_tester.cc` to verity this PR. And make `test_analyzer_transformer` runs in `SERIAL` in `CMAKELISTS.txt`.

**Comment by jczaja** [comment]
@luotao1 I have updated changes with your suggestions.

File: paddle/fluid/inference/tests/api/analyzer_transformer_tester.cc [hunk 1]
```diff
@@ -183,17 +183,27 @@ void SetInput(std::vector<std::vector<PaddleTensor>> *inputs) {
...
 // Easy for profiling independently.
-TEST(Analyzer_Transformer, profile) {
+void profile(bool use_mkldnn = false) {
   AnalysisConfig cfg;
   SetConfig(&cfg);
   std::vector<PaddleTensor> outputs;
+  if (use_mkldnn) {
+    cfg.EnableMKLDNN();
+    std::unordered_set<std::string> op_list = {"transpose2"};
+    cfg.SetMKLDNNOp(op_list);
```
Comment (luotao1) [inline paddle/fluid/inference/tests/api/analyzer_transformer_tester.cc]: Why should we specify op_list for transformer ut? Could we remove line192-193?

Comment (jczaja) [reply paddle/fluid/inference/tests/api/analyzer_transformer_tester.cc]: @luotao1 Good point. Manual selection of mkl-dnn ops for transformer was removed.

**Review by luotao1 — APPROVED** [review]
LGTM! Thanks very much!

**Pull request merged**